sklearn/gaussian_process/tests/test_kernels.py

```diff
@@ -51,8 +51,10 @@
 
 @pytest.mark.parametrize('kernel', kernels)
 def test_kernel_gradient(kernel):
+    def create_kernel(kernel, X):
+        return kernel(X, eval_gradient=True)
     # Compare analytic and numeric gradient of kernels.
-    K, K_gradient = kernel(X, eval_gradient=True)
+    K, K_gradient = assert_no_warnings(create_kernel, kernel, X)
```

Comment:
You should be able to use `eval_gradient` as a keyword directly, that is:
```suggestion
    K, K_gradient = assert_no_warnings(kernel, X, eval_gradient=True)
```

In that case, you can delete `create_kernel`.